src/routes/home.js

```diff
@@ -37,3 +38,14 @@ module.exports = {
     }
   }
 };
+
+const getUserData = (room, endpoint) => {
+  let roomId = Rooms[room];
+  let endpointId = Rooms[room][endpoint];
+
+  let roomName = roomId.getroomName();
+  let username = roomId.endpoints[endpointId].getUsername();
+  let permissions = roomId.endpoints[endpointId].getPermissions();
```

Comment:
As above, this getPermissions method would be called on the room object itself...
So-
```javascript
roomId.getPermissions(endpointId);
```